networkx/algorithms/flow/tests/test_networksimplex.py

```diff
@@ -0,0 +1,401 @@
+import pytest
+import networkx as nx
+import os
+
+
+class TestNetworkSimplex:
+    def get_flowcost_from_flowdict(self, G, flowDict):
+        """Returns flow cost calculated from flow dictionary"""
+        flowCost = 0
+        for u in flowDict.keys():
+            for v in flowDict[u].keys():
+                flowCost += flowDict[u][v] * G[u][v]["weight"]
+        return flowCost
+
+    def test_infinite_demand_raise(self):
+        G = nx.DiGraph()
+        inf = float("inf")
+        G.add_node("a", demand=inf)
+        G.add_node("b", demand=-5)
+        G.add_node("c", demand=50000000)
+        G.add_node("d", demand=-49999995)
+        G.add_edge("a", "b", weight=3, capacity=4)
+        G.add_edge("a", "c", weight=6, capacity=10)
+        G.add_edge("b", "d", weight=1, capacity=9)
+        G.add_edge("c", "d", weight=2, capacity=5)
+        pytest.raises(nx.NetworkXError, nx.network_simplex, G)

```

Comment:
It seems like this same basic example graph (with minor modifications) is used for the next 6 or so tests. The redundancy could really be cut down by wrapping the basic graph generation up in a fixture, and passing that into each of the tests; e.g. something like:

```suggestion
    @pytest.fixture
    def simple_flow_graph():
        G = nx.DiGraph()
        G.add_node("a", demand=5)
        G.add_node("b", demand=-5)
        G.add_node("c", demand=50000000)
        G.add_node("d", demand=-49999995)
        G.add_edge("a", "b", weight=3, capacity=4)
        G.add_edge("a", "c", weight=6, capacity=10)
        G.add_edge("b", "d", weight=1, capacity=9)
        G.add_edge("c", "d", weight=2, capacity=5)
        return G

    def test_infinite_demand_raise(self, simple_flow_graph):
        G = simple_flow_graph
        inf = float("inf")
        nx.set_node_attributes(G, {"a": {"demand": inf}})
        pytest.raises(nx.NetworkXError, nx.network_simplex, G)

```

Then the next few tests can be rewritten to use the fixture in a similar manner.